src/Data/Functor/Yoneda.hs

```diff
@@ -138,24 +139,67 @@ instance Adjunction f g => Adjunction (Yoneda f) (Yoneda g) where
   unit = liftYoneda . fmap liftYoneda . unit
   counit (Yoneda m) = counit (m lowerYoneda)
 
--- instance Show1 f => Show1 (Yoneda f) where
+instance Show1 f => Show1 (Yoneda f) where
+#if LIFTED_FUNCTOR_CLASSES
+  liftShowsPrec sp sl d (Yoneda f) =
+    showsUnaryWith (liftShowsPrec sp sl) "liftYoneda" d (f id)
+#else
+  showsPrec1 d (Yoneda f) = showParen (d > 10) $
+    showString "liftYoneda " . showsPrec1 11 (f id)
+#endif
+
+instance (Read1 f, Functor f) => Read1 (Yoneda f) where
+#if LIFTED_FUNCTOR_CLASSES
+  liftReadsPrec rp rl = readsData $
+    readsUnaryWith (liftReadsPrec rp rl) "liftYoneda" liftYoneda
+#else
+  readsPrec1 d = readParen (d > 10) $ \r' ->
+    [ (liftYoneda f, t)
+    | ("liftYoneda", s) <- lex r'
+    , (f, t) <- readsPrec1 11 s
+    ]
+#endif
+
 instance Show (f a) => Show (Yoneda f a) where
```

Comment:
Did you intend to change this instance to use `Show1 f` instead of `Show (f a)`?

Reply:
My reasoning was that "if the expression would need `Functor f` (as in this case) then `Show1 f` is enough. 
So here we could have `Show1 f, Show a` context.

In `Coyoneda` case `liftYoneda` can lift anything, so there I'd rather stick with `Show (f a)`.

---

In other ways: if `f` can be some weird *GADT* then `Show (f a)` is better, as it will cover more cases. But if its ordinary ADT, then it's a `Functor`, so most likely, `Show (f a)` requires `Show a` context too.

---

Third way to say: I try not to exclude possible use cases where `Show (f a)` doesn't need `Show a`.